server/src/main/java/org/elasticsearch/search/fetch/subphase/highlight/AbstractHighlighterBuilder.java

```diff
@@ -193,9 +189,6 @@ public final void writeTo(StreamOutput out) throws IOException {
         }
         out.writeOptionalWriteable(order);
         out.writeOptionalBoolean(highlightFilter);
-        if (out.getTransportVersion().before(TransportVersions.V_8_8_0)) {
-            out.writeOptionalBoolean(false);
-        }
```

Comment:
I have not added a new transport version associated with this change. My assumption is that it is not needed given that 9.0 will never communicate with versions previous than 8.last, hence the conditionals under which the flag was read or written are always going to render false.